### PR TITLE
Add ES CSS Modules

### PR DESCRIPTION
**Pull request by jacobp100**
Not 100% sure it’s applicable here, but it’s very similar to React CSS Modules.

### CODE REVIEW

**Comment by MicheleBertoli** [comment]
Hey @jacobp100, thanks for you PR.
Any CSS in JS library is welcome here :)
The only problem is that the demo should be consistent with the existing ones.
Can you please update it?

**Comment by jacobp100** [comment]
Should be consistent now!

File: README.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ For example, if a package supports the css file extraction you can run the autop
 | [css-ns](https://github.com/jareware/css-ns) | 1.0.0 | | x | x | | x |
 | [cssx-loader](https://github.com/krasimir/cssx) | 3.8.0 | x | x | x | | x |
 | [hyperstyles](https://github.com/colingourlay/hyperstyles) | 3.3.0 | | x | x | | x |
-| [j2c](https://github.com/j2css/j2c) | 0.10.0 | | x | x | x | x |
+| [es-css-modules](https://github.com/jacobp100/es-css-modules) | 1.2.3 | | x | x | | x |
```
Comment (MicheleBertoli) [inline README.md]: `es-css-modules` should be before `hyperstyles`

File: README.md [hunk 2]
```diff
@@ -27,7 +27,8 @@ For example, if a package supports the css file extraction you can run the autop
...
+| [j2c](https://github.com/j2css/j2c) | 0.10.0 | | x | x | | x |
```
Comment (MicheleBertoli) [inline README.md]: `j2c` should be untouched

File: es-css-modules/package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "gulp-es-css-modules-demo",
+  "version": "0.0.1",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (MicheleBertoli) [inline es-css-modules/package.json]: there should be a "build" script to create the bundles.

File: es-css-modules/index.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>es-css-modules - CSS in JS</title>
+    <link rel="stylesheet" href="bundle.css">
```
Comment (MicheleBertoli) [inline es-css-modules/index.html]: `bundle.css` doesn't exist after running `gulp` (the path is `dist/button.css`).

File: es-css-modules/index.html [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  </head>
+  <body>
+    <div id="content"></div>
+    <script src="bundle.js"></script>
```
Comment (MicheleBertoli) [inline es-css-modules/index.html]: `bundle.js` doesn't exists, it should be created (with `babel` or `webpack`) during the build process.

**Comment by MicheleBertoli** [comment]
Thank you very much @jacobp100.
Where are almost there, just a few inline comments.

File: README.md (hunk 2 above)
Comment (jacobp100) [inline README.md]: Good catch!

**Comment by MicheleBertoli** [comment]
Thank you very much @jacobp100.

Now, when I run `npm run build && open index.html` I get the following error in my browser console:
> Uncaught Error: Cannot find module "./button.m.css"

If I run it again it works but then I get:
>  React.render is deprecated. Please use ReactDOM.render from require('react-dom') instead.

The project should work according to the instructions in the [usage](https://github.com/MicheleBertoli/css-in-js#usage) section.
Can you please fix the `gulp` issue and use `react-dom` to render the button?

**Comment by jacobp100** [comment]
Fixed those issues!

**Comment by MicheleBertoli** [comment]
Awesome, thanks @jacobp100!

**Comment by jacobp100** [comment]
Thanks for merging!

**Pull request merged**